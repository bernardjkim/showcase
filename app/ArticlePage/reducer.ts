/*
 *
 * ArticlePage reducer
 *
 */

import { ArticleActionTypes, ArticleState, SET_ARTICLE_ID } from './types';

// The initial state of the App
export const initialState: ArticleState = {
  articleId: undefined,
};

function articlePageReducer(state: ArticleState = initialState, action: ArticleActionTypes) {
  switch (action.type) {
    case SET_ARTICLE_ID:
      return {
        ...state,
        articleId: action.articleId,
      };

    default:
      return state;
  }
}

export default articlePageReducer;
