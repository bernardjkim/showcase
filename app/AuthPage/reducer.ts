/*
 *
 * AuthPage reducer
 *
 */

// import { LOCATION_CHANGE } from 'react-router-redux';
import {
  AuthState,
  AuthActionTypes,
  CLEAR_ERRORS,
  CREATE_TOKEN,
  CREATE_TOKEN_ERROR,
  CREATE_TOKEN_SUCCESS,
  CREATE_USER,
  CREATE_USER_ERROR,
  CREATE_USER_SUCCESS,
} from './types';

export const initialState: AuthState = {
  loading: false,
  error: undefined,
};

function AuthPageReducer(
  state: AuthState = initialState,
  action: AuthActionTypes,
) {
  switch (action.type) {
    // case LOCATION_CHANGE:
    //   return { ...state, error: undefined };

    case CLEAR_ERRORS:
      return { ...AbortController, error: undefined };

    case CREATE_TOKEN:
      return { ...state, loading: true, error: undefined };

    case CREATE_TOKEN_SUCCESS:
      return { ...state, loading: false };

    case CREATE_TOKEN_ERROR: {
      return { ...state, loading: false, error: { createToken: action.error } };
    }

    case CREATE_USER:
      return { ...state, loading: true, error: undefined };

    case CREATE_USER_SUCCESS:
      return { ...state, loading: false };

    case CREATE_USER_ERROR:
      return { ...state, loading: false, error: { createUser: action.error } };

    default:
      return state;
  }
}

export default AuthPageReducer;
