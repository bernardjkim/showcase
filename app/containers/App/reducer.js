/*
 * 
 * App reducer
 * 
 */

import { fromJS } from 'immutable';
import {
  CREATE_TOKEN,
  CREATE_TOKEN_ERROR,
  CREATE_TOKEN_SUCCESS,
} from './constants';

// The initial state of the App
export const initialState = fromJS({
  loading: false,
  error: false,
  token: false,
});

function appReducer(state = initialState, action) {
  switch (action.type) {
    case CREATE_TOKEN:
      return state.set('loading', true).set('error', false);

    case CREATE_TOKEN_SUCCESS:
      return state.set('token', action.token).set('loading', false);

    case CREATE_TOKEN_ERROR:
      return state.set('error', fromJS(action.error)).set('loading', false);

    default:
      return state;
  }
}

export default appReducer;
