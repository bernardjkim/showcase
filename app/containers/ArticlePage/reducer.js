/*
 *
 * ArticlePage reducer
 *
 */

import { fromJS } from 'immutable';
import {
  CREATE_COMMENT,
  CREATE_COMMENT_ERROR,
  CREATE_COMMENT_SUCCESS,
  LIKE_ARTICLE,
  LIKE_ARTICLE_ERROR,
  LIKE_ARTICLE_SUCCESS,
  LOAD_ARTICLE,
  LOAD_ARTICLE_ERROR,
  LOAD_ARTICLE_SUCCESS,
  LOAD_COMMENTS,
  LOAD_COMMENTS_ERROR,
  LOAD_COMMENTS_SUCCESS,
} from './constants';

// The initial state of the App
export const initialState = fromJS({
  loading: false,
  error: false,
  /**
   * article
   * @field {string}  _id         - Article id
   * @field {string}  uri         - Article uri
   * @field {string}  github      - GitHub repo
   * @field {string}  description - Article description
   * @field {string}  image       - Article image link
   * @field {number}  likes       - Number of likes
   * @field {bool}    likedByUser - Article is liked by user
   * @field {array}   comments    - Array of comments
   */
  article: false,
});

function articlePageReducer(state = initialState, action) {
  switch (action.type) {
    case CREATE_COMMENT:
      return state.set('loading', true).set('error', false);

    case CREATE_COMMENT_SUCCESS:
      return state
        .updateIn(['article', 'comments'], comments =>
          comments.push(fromJS(action.comment)),
        )
        .set('loading', false);

    case CREATE_COMMENT_ERROR:
      return state.set('error', fromJS(action.error)).set('loading', false);

    case LIKE_ARTICLE:
      return state.set('loading', true).set('error', false);

    case LIKE_ARTICLE_SUCCESS:
      return state
        .updateIn(['article', 'likes'], likes => likes + 1)
        .setIn(['article', 'likedByUser'], true)
        .set('loading', false);

    case LIKE_ARTICLE_ERROR:
      return state.set('error', fromJS(action.error)).set('loading', false);

    case LOAD_ARTICLE:
      return state
        .set('loading', true)
        .set('error', false)
        .set('article', false);

    case LOAD_ARTICLE_SUCCESS:
      return state.set('article', fromJS(action.article)).set('loading', false);

    case LOAD_ARTICLE_ERROR:
      return state.set('error', fromJS(action.error)).set('loading', false);

    case LOAD_COMMENTS:
      return state.set('loading', true).set('error', false);

    case LOAD_COMMENTS_SUCCESS:
      return state.setIn(['article', 'comments'], fromJS(action.comments));

    case LOAD_COMMENTS_ERROR:
      return state.set('error', fromJS(action.error)).set('loading', false);

    default:
      return state;
  }
}

export default articlePageReducer;
