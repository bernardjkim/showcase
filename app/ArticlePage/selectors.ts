import { createSelector } from 'reselect';
import { initialState } from './reducer';
import { ArticleState } from './types';

/**
 * Direct selector to the articlePage state domain
 */

const selectArticlePageDomain = (state: any): ArticleState => state.get('articlePage', initialState);

/**
 * Other specific selectors
 */

const makeSelectLoading = () =>
  createSelector(
    selectArticlePageDomain,
    articleState => articleState.loading,
  );

const makeSelectError = () =>
  createSelector(
    selectArticlePageDomain,
    articleState => articleState.error,
  );

const makeSelectArticle = () =>
  createSelector(
    selectArticlePageDomain,
    articleState => articleState.article,
  );

const makeSelectComments = () =>
  createSelector(
    selectArticlePageDomain,
    articleState => articleState.comments,
  );

const makeSelectLikes = () =>
  createSelector(
    selectArticlePageDomain,
    articleState => articleState.likes,
  );
/**
 * Default selector used by ArticlePage
 */

const makeSelectArticlePage = () =>
  createSelector(
    selectArticlePageDomain,
    substate => substate,
  );

export default makeSelectArticlePage;
export {
  selectArticlePageDomain,
  makeSelectArticle,
  makeSelectComments,
  makeSelectLikes,
  makeSelectLoading,
  makeSelectError,
};
