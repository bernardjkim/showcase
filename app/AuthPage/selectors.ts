import { createSelector } from 'reselect';
import { initialState } from './reducer';
import { AuthState } from './types';

/**
 * Direct selector to the AuthPage state domain
 */

const selectAuthPageDomain = (state: any): AuthState =>
  state.get ? state.get('authPage', initialState) : initialState;

/**
 * Other specific selectors
 */
const makeSelectLoading = () =>
  createSelector(
    selectAuthPageDomain,
    authState => authState.loading,
  );

const makeSelectError = () =>
  createSelector(
    selectAuthPageDomain,
    authState => authState.error,
  );

/**
 * Default selector used by AuthPage
 */

const makeSelectAuthPage = () =>
  createSelector(
    selectAuthPageDomain,
    substate => substate,
  );

export default makeSelectAuthPage;
export { selectAuthPageDomain, makeSelectError, makeSelectLoading };
