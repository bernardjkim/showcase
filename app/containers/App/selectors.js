import { createSelector } from 'reselect';
import { initialState } from './reducer';

const selectRouter = state => state.get('router');

const makeSelectLocation = () =>
  createSelector(selectRouter, routerState =>
    routerState.get('location').toJS(),
  );

/**
 * Direct selector to the global state domain
 */

const selectGlobalDomain = state => state.get('global', initialState);

/**
 * Other specific selectors
 */

const makeSelectLoading = () =>
  createSelector(selectGlobalDomain, globalState => globalState.get('loading'));

const makeSelectError = () =>
  createSelector(selectGlobalDomain, globalState => globalState.get('error'));

const makeSelectUser = () =>
  createSelector(selectGlobalDomain, globalState => globalState.get('user'));

const makeSelectValidateToken = () =>
  createSelector(selectGlobalDomain, globalState =>
    globalState.get('validateToken'),
  );

/**
 * Default selector used by App
 */

const makeSelectApp = () =>
  createSelector(selectGlobalDomain, substate => substate.toJS());

export default makeSelectApp;
export {
  makeSelectLocation,
  selectGlobalDomain,
  makeSelectUser,
  makeSelectValidateToken,
  makeSelectLoading,
  makeSelectError,
};
