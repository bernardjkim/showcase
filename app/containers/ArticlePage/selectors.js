import { createSelector } from 'reselect';
import { initialState } from './reducer';

/**
 * Direct selector to the articlePage state domain
 */

const selectArticlePageDomain = state => state.get('articlePage', initialState);

/**
 * Other specific selectors
 */

const makeSelectLoading = () =>
  createSelector(selectArticlePageDomain, articleState =>
    articleState.get('loading'),
  );

const makeSelectError = () =>
  createSelector(selectArticlePageDomain, articleState =>
    articleState.get('error'),
  );

const makeSelectArticle = () =>
  createSelector(selectArticlePageDomain, articleState =>
    articleState.get('article'),
  );

/**
 * Default selector used by ArticlePage
 */

const makeSelectArticlePage = () =>
  createSelector(selectArticlePageDomain, substate => substate.toJS());

export default makeSelectArticlePage;
export {
  selectArticlePageDomain,
  makeSelectArticle,
  makeSelectLoading,
  makeSelectError,
};
