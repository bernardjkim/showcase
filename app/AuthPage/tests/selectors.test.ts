import { initialState } from '../reducer';
import { makeSelectError, makeSelectLoading, selectAuthPageDomain } from '../selectors';

describe('selectAuthPageDomain', () => {
  it('should select the auth page state', () => {
    expect(selectAuthPageDomain(initialState)).toMatchSnapshot();
  });
});

describe('makeSelectLoading', () => {
  const loadingSelector = makeSelectLoading();
  it('should select the loading state', () => {
    expect(loadingSelector(initialState)).toMatchSnapshot();
  });
});

describe('makeSelectError', () => {
  const errorSelector = makeSelectError();
  it('should select the error state', () => {
    expect(errorSelector(initialState)).toMatchSnapshot();
  });
});
