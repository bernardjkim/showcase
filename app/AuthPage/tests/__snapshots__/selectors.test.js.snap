// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeSelectError should select the error state 1`] = `false`;

exports[`makeSelectLoading should select the loading state 1`] = `false`;

exports[`selectAuthPageDomain should select the auth page state 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
}
`;
